### PR TITLE
Generate input types (and more fixes)

### PR DESCRIPTION
**Pull request by fehnomenal**
See https://github.com/HoudiniGraphql/houdini/discussions/475#discussioncomment-3467838

I also unified and fixed some things I stumbled over.

Does this need a changeset?

### CODE REVIEW

**Comment by AlecAivazis** [comment]
No, I dont think this one needs a changeset since the user isn't affected

File: src/cmd/generators/stores/mutation.ts [hunk 1]
```diff
@@ -29,10 +29,13 @@ export const ${globalStoreName} = new ${storeName}()
 export default ${globalStoreName}
 `
 
-	// type definitions
```
Comment (AlecAivazis) [inline src/cmd/generators/stores/mutation.ts:32]: I don't think this type is always defined by the typescript generator. Here is the test for a query with no inputs: https://github.com/HoudiniGraphql/houdini/blob/directory-routes/src/cmd/generators/typescript/typescript.test.ts#L233.

Fwiw, I agree it's simpler if we do define the type so I think we just need to tweak the typescript generator and then this should be good to go

**Review by AlecAivazis — CHANGES_REQUESTED** [review]

File: src/cmd/generators/stores/mutation.ts (hunk 1 above)
Comment (fehnomenal) [reply src/cmd/generators/stores/mutation.ts:32]: Sorry, I don't get it :sweat_smile: 

The updated test is here and contains the generated input type: https://github.com/fehnomenal/houdini/blob/always-generate-input-types/src/cmd/generators/typescript/typescript.test.ts#L233

The generation code lives in `typescript/index.ts`.

Comment (fehnomenal) [reply src/cmd/generators/stores/mutation.ts:32]: Ahh, this was just a change to get the prelude to the typescript code in the same shape for queries, mutations, fragments and subscriptions.

Comment (AlecAivazis) [reply src/cmd/generators/stores/mutation.ts:32]: Ohh you're totally right! Idk how I missed that.

**Review by AlecAivazis — APPROVED** [review]
Okay, found some time to pull this down and verify it. Thanks again!

**Pull request merged**